docs/dev/reference/dca/callbacks.md

```diff
@@ -431,4 +431,50 @@ a button for an import "wizard".
 **return:** `string` HTML for the button
 {{% /expand %}}
 
+
+***
+
+## Examples of defining callbacks
+
+You can register any of the abovementioned callbacks by using annotations:
+
+```php
+// src/Controller/EventListener/DataContainer/MyTableListener.php
+
+namespace App\EventListener\DataContainer;
+
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\DataContainer;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+class MyTableListener implements ServiceAnnotationInterface
+{
+
+    /**
+     * @Callback(table="tl_my_table", target="fields.xxMyField.options")
+     */
+    public function onOptionsCallback(DataContainer $dc): array
+    {
+        return ['test1', 'test2'];
+    }
+}
+```
+
+When [autowiring][autowiring] is enabled in your app, your callback is registered successfully.
+
+In case you don't use autowiring, you need to tag the corresponding service of your callback
+class:
+
+```yml
+// config/services.yml
+
+App\EventListener\DataContainer\MyTableListener:
+  tags:
+    - { name: contao.callback, table: tl_my_table, target: fields.xxMyField.options [, method: onOptionsCallback] }
```

Comment:
Do you mean auto configuration? Still, you could use the service annotation by tagging the service with `terminal42_service_annotation`.

I think it should be something like this: 
* You can define the callback directly as a tag of your service: ...
* Alternatively you can use the callback service annotation to do so: [...] 
Note: If autowiring is disabled you need to add a tag `['terminal42_service_annotation']` to your service instead of the `ServiceAnnotationInterface` marker interface in order to use the callback annotation.

Reply:
Yes, autoconfiguration not autowiring :)
And I think we should add a hint that it was only introduced in version 4.8 - I think? :) Not sure.

Reply:
I'll clarify that there are three approaches:
1. tagged service, no annotation
2. tagged service and terminal42_service_annotation annotation
3. annotation and autoconfiguration

Correct?

> And I think we should add a hint that it was only introduced in version 4.8 

I'm aware of this but information about supported versions is lacking everywhere in the documentation. I guess we assume developers to use the latest version @fritzmg ?

Reply:
Hm? No it's actually full of `{{< version "4.5" >}}` and the like hints :)

Reply:
@richardhj I think 1) should be the default - 2) & 3) are basically the same thing but with sevice annotations as a helper. 

@fritzmg I'm not sure how to deal with this in the docs in general. Maybe we would need a table/swicher widget for some things like symfony has for their various ways of writing config (yaml|xml|php)?

Reply:
When I looked at the docs for front end modules / content elements and hooks, AFAIR there was no note about the version requirements

Reply:
> @richardhj I think 1) should be the default - 2) & 3) are basically the same thing but with sevice annotations as a helper.

We should add a general section about autoconfiguration then in the docs?

Reply:
> When I looked at the docs for front end modules / content elements and hooks, AFAIR there was no note about the version requirements

There definitely is for Hooks. See https://docs.contao.org/dev/framework/hooks/#using-annotations for example.